src/runtime/storageNG/tests/reference-mode-store-test.ts

```diff
@@ -0,0 +1,362 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {assert} from '../../../platform/chai-web.js';
+import {Store, StorageMode, ProxyMessageType} from '../store.js';
+import {Exists, DriverFactory} from '../drivers/driver-factory.js';
+import {CRDTCount, CountOpTypes, CountData, CountOperation} from '../../crdt/crdt-count.js';
+import {StorageKey} from '../storage-key.js';
+import {DirectStore} from '../direct-store.js';
+import {MockStorageKey, MockStorageDriverProvider, MockDriver, MockHierarchicalStorageKey} from '../testing/test-storage.js';
+import {ReferenceModeStore, ReferenceCollection, Reference} from '../reference-mode-store.js';
+import {CRDTEntity, EntityOpTypes} from '../../crdt/crdt-entity.js';
+import {CRDTCollection, CollectionOpTypes, CollectionData, CollectionOperation} from '../../crdt/crdt-collection.js';
+import {CRDTSingleton} from '../../crdt/crdt-singleton.js';
+
+let testKey: StorageKey;
+
+class MyEntityModel extends CRDTEntity<{name: {id: string}, age: {id: string, value: number}}, {}> {
+  constructor() {
+    super({name: new CRDTSingleton<{id: string}>(), age: new CRDTSingleton<{id: string, value: number}>()}, {});
+  }
+}
+
+class MyEntity {
+  id: string;
+  name: {id: string};
+  age: {id: string, value: number};
+}
+
+class MyEntityCollection extends CRDTCollection<MyEntity> {}
+
+describe('Reference Mode Store', async () => {
+
+  beforeEach(() => {
+    testKey = new MockHierarchicalStorageKey();
+    DriverFactory.clearRegistrationsForTesting();
+  });
+
+  after(() => {
+    DriverFactory.clearRegistrationsForTesting();
+  });
+
+  /*
+  it(`will throw an exception if an appropriate driver can't be found`, async () => {
+    const store = new Store(testKey, Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    try {
+      await store.activate();
+      assert.fail('store.activate() should not have succeeded');
+    } catch (e) {
+      assert.match(e.toString(), /No driver exists/);
+    }
+  });
+  */
+
+  /*
+  it('will construct Direct stores when required', async () => {
+    DriverFactory.register(new MockStorageDriverProvider());
+
+    const store = new Store(testKey, Exists.ShouldCreate, null, StorageMode.Direct, CRDTCount);
+    const activeStore = await store.activate();
+
+    assert.equal(activeStore.constructor, DirectStore);
+  });
+  */
```

Comment:
Did you mean to leave these commented out?

Reply:
I should skip them instead. I haven't set up the construction pathway for this yet and these tests rely on that being there.